### PR TITLE
InvocationFuture minor improvements

### PR DESCRIPTION
**Pull request by pveentjer**
1: always using System.currentTimeMillis instead of Clock at some parts and System at others
2: improved time toString (a shorter but easier to read time)

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/util/StringUtil.java [hunk 1]
```diff
@@ -127,4 +130,17 @@ public static String lowerCaseInternal(String s) {
...
+    /**
+     * Returns a String representation of the time.
+     *
+     * This method is not particularly efficient since it generates a ton of litter.
+     *
+     * @param timeMillis time in millis
+     * @return the String
+     */
+    public static String timeToString(long timeMillis) {
+        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: It's a shame that `SimpleDateFormat` is not thread-safe, would save some litter :(

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: It isn't only the litter of the DateFormat object, but also during the format method, waste is generated. For its current purposes it is fine :)

**Comment by pveentjer** [comment]
run-lab-run

**Comment by pveentjer** [comment]
verify

**Comment by Donnerbart** [comment]
Judging `Clock.currentTimeMillis()` vs. `System.currentTimeMillis()` always gives me a hard time.

The changes in `newOperationTimeoutException()` should be fine, since it should align the timestamps with the ones in the logs, which should make comparisons easier. It's anyway just for log output, so it should not have any critical impact.

The original code in `runAsynchronous()` looks broken anyway, since both clocks were used for the same variable. So with a big offset in `Clock` a `TimeoutException` could be thrown immediately, if I'm not wrong. So the change should fix this. Since this is anyway about a relative time comparison, the `System` clock should be fine.

**Comment by pveentjer** [comment]
In production Clock and System time are exactly the same; however for testing you can tinker with the clock to skew.

**Comment by Donnerbart** [comment]
Okay, I just looked this up. I though `Clock` was the same as `ClusterClock`, but apparently those are two different things.

**Pull request merged**